### DISCUSSION TITLE
[Implemented] Should `{tlf}` be a hard or a soft dependency?

### DISCUSSION THREAD

**Discussion by IndrajeetPatil** [Open-Systems-Pharmacology/OSPSuite-R#921] [General]
@Yuri05 had mentioned the following:

@IndrajeetPatil @PavelBal @msevestre 
Do we want to have `tlf` and `patchwork` as **mandatory** prerequisites (like now) or is it sufficient to have them as **optional** prereqs? 
I would vote for **optional**, because - except for plots - `ospsuite` is still fully functional without those packages installed.

_Originally posted by @Yuri05 in https://github.com/Open-Systems-Pharmacology/OSPSuite-R/pull/882#discussion_r855506637_

---

1. This would mean that none of the functions from `{tlf}` can be in the _ospsuite_ `NAMESPACE` and the plot grid functions/classes can't be re-exported, like so:

```r
#' @export
#' @importFrom tlf plotGrid
tlf::plotGrid

#' @export
#' @importFrom tlf PlotGridConfiguration
tlf::PlotGridConfiguration
```

2. This also increases the code complexity because we will need to check if _tlf_ package is installed at the top level every time _tlf_ is used in code, examples, tests, vignettes, etc.

Just wanted to make sure that the team was on-board with this before I made the necessary changes.

**Comment by PavelBal** [comment]
Actually I am for mandatory. I would hate it to suddenly realize that some functionality of the package does not work because a dependency is optional.

**Comment by Yuri05** [comment]
ok, let's make it mandatory then
